ui/src/components/credentials/IssueCredentialForm.tsx

```diff
@@ -27,24 +54,73 @@ export function IssueCredentialForm({
   schema: Schema;
   type: "directIssue" | "credentialLink";
 }) {
-  const rawCredentialSubjectAttributes =
-    (jsonSchema.type === "object" &&
-      jsonSchema.schema.properties
-        ?.filter((child): child is ObjectAttribute => child.type === "object")
-        .find((child) => child.name === "credentialSubject")?.schema.properties) ||
-    null;
-
-  const credentialSubjectAttributes =
-    rawCredentialSubjectAttributes &&
-    rawCredentialSubjectAttributes.filter((attribute) => attribute.name !== "id");
-
-  return credentialSubjectAttributes ? (
+  const [inputErrors, setInputErrors] = useState<InputErrors>();
+
+  function isFormValid(value: Record<string, unknown>, objectAttribute: ObjectAttribute): boolean {
+    const serializedSchemaForm = serializeSchemaForm({
+      attribute: makeAttributeOptional(objectAttribute),
+      value,
+    });
+    if (serializedSchemaForm.success) {
+      const { properties, required, type } = objectAttribute.schema;
+      try {
+        const ajv = new Ajv({ allErrors: true });
+        addFormats(ajv);
+        const validate = ajv.compile({
+          properties,
+          required,
+          type,
+        });
+        const valid = validate(serializedSchemaForm.data);
+
+        if (valid) {
+          setInputErrors(undefined);
+          return true;
+        } else if (validate.errors) {
+          setInputErrors(
+            validate.errors.reduce((acc: InputErrors, curr: ErrorObject): InputErrors => {
+              if (curr.keyword === "required") {
+                // filtering out required errors since we manage these from the antd form
+                return acc;
+              } else {
+                const errorMsg = curr.message
+                  ? curr.message.charAt(0).toUpperCase() + curr.message.slice(1)
+                  : "Unknown validation error";
+                const path = curr.instancePath
+                  .split("/")
+                  .filter((segment) => segment !== "/" && segment !== "");
+                return addErrorToPath(acc, path, errorMsg);
+              }
+            }, {})
+          );
+        }
+      } catch (error) {
+        notifyError(buildAppError(error));
+      }
+    }
+    return false;
```

Comment:
There's a hell of a bug here. I haven't been able to find a solution yet.

The problem is related to antd components (InputNumber and Date specifically, but there might be more cases) returning `null` when the value is removed.

This is easy to fix today by transforming all nulls to undefined, but we need to find another solution because this does interfere with the fact that we need to support `null` attributes.